config/passport.js

```diff
@@ -0,0 +1,28 @@
+const passport = require("passport");
+const passportJWT = require("passport-jwt");
+const ExtractJWT = passportJWT.ExtractJwt;
+const Strategy = passportJWT.Strategy;
+
+require("dotenv").config();
+const SECRET = process.env.SECRET_KEY;
+const User = require("../models/user");
+
+const params = {
+  secretOrKey: SECRET,
+  jwtFromRequest: ExtractJWT.fromAuthHeaderAsBearerToken(),
+};
+
+passport.use(
+  new Strategy(params, async (payload, done) => {
+    try {
+      const user = await User.findOne({ _id: payload.id });
+
+      if (!user) {
+        return done(new Error("User not found"));
+      }
+      return done(null, user);
```

Comment:
tutaj pewnie mozna ladnie:
```js
return user
? done(null, user)
: done(new Error("User not found"));
```

Diabli wiedza, dlaczego oficjalne przyklady dalej bawia sie w takie "zacznijmy od zaprzeczenia" if'y. Generalnie, zawsze lepiej zaczac (jesli tylko sie da (ale tutaj sie totalnie da)) od "jesli prawda", anizeli "jesli nie prawda", bo latwiej nam to rozumiec.